app/src/main/java/fr/free/nrw/commons/nearby/NearbyMapFragment.java

```diff
@@ -385,10 +390,65 @@ private void passInfoToSheet(Place place) {
         commonsButton.setEnabled(place.hasCommonsLink());
         commonsButton.setOnClickListener(view -> openWebView(place.siteLinks.getCommonsLink()));
 
-        icon.setImageResource(place.getDescription().getIcon());
-        description.setText(place.getDescription().getText());
-        title.setText(place.name);
-        distance.setText(place.distance);
+        icon.setImageResource(place.getLabel().getIcon());
+
+        description.setText(place.getLongDescription());
+        title.setText(place.name.toString());
+        distance.setText(place.distance.toString());
+
+        fabCamera.setOnClickListener(view -> {
+            Timber.d("Camera button tapped. Image title: " + place.getName() + "Image desc: " + place.getLongDescription());
+            controller = new ContributionController(this);
+            DirectUpload directUpload = new DirectUpload(place.getName(), place.getLongDescription(), this, controller);
+            directUpload.storeSharedPrefs();
+            directUpload.initiateCameraUpload();
+        });
+
+        fabGallery.setOnClickListener(view -> {
+            Timber.d("Gallery button tapped. Image title: " + place.getName() + "Image desc: " + place.getLongDescription());
+            controller = new ContributionController(this);
+            DirectUpload directUpload = new DirectUpload(place.getName(), place.getLongDescription(), this, controller);
+            directUpload.storeSharedPrefs();
+            directUpload.initiateGalleryUpload();
+        });
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        Timber.d("onRequestPermissionsResult: req code = " + " perm = " + permissions + " grant =" + grantResults);
+
+        switch (requestCode) {
+            // 1 = "Read external storage" allowed when gallery selected
+            case 1: {
+                if (grantResults.length > 0 && grantResults[0] == PERMISSION_GRANTED) {
+                    Timber.d("Call controller.startGalleryPick()");
+                    controller.startGalleryPick();
+                }
+            }
+            break;
+            
+            // 3 = "Write external storage" allowed when camera selected
+            case 3: {
```

Comment:
Am keeping this as "3" in line with the request codes used at ContributionsListFragment.java to prevent confusion

Reply:
I'm not a fan of magic numbers, but the comment helps.